include/oneapi/dpl/pstl/hetero/utils_hetero.h

```diff
@@ -117,11 +117,9 @@ struct __identity_reduce_fn
         using ::std::get;
         auto __chosen_for_min = __a;
         auto __chosen_for_max = __b;
-        // if b "<" a or if b "==" a and b_index < a_index
-        if (__comp(get<2>(__b), get<2>(__a)) || (!__comp(get<2>(__a), get<2>(__b)) && get<0>(__b) < get<0>(__a)))
+        if (__comp(get<2>(__b), get<2>(__a)))
```

Comment:
Probably does it make sense to add a couple of  asserts like "asset(a_index < b_index)" here  and prior the second if?

Reply:
Thanks - good suggestion :+1:

Reply:
The assertions do not use the comparator but `operator<`, which seems a mistake.

Reply:
I think the comparator should be used on the values only. The assertions check if the indices are ordered such that we can leave out the special case if both values are the same.